include/picongpu/fields/laserProfiles/laserExpRampWithPrepulse.hpp

```diff
@@ -0,0 +1,188 @@
+            //    throw std::invalid_argument("\n\nAttention, the intensities of the ramp are very large, the extrapolation to the time of the main pulse would give more than 50% of the pulse amplitude - this is not a gaussian pulse at all anymore, probably something wrong?!\n");
+
+            env += AMPLITUDE * (1.-ramp_when_peakpulse) * gauss(runTime-endUpramp);
+            env += AMP_PREPULSE * gauss(runTime - TIME_PREPULSE);
+            if (during_first_exp)
+                env += extrapolate(
+                    TIME_1, 
+                    AMP_1, 
+                    TIME_2, 
+                    AMP_2, 
+                    runTime
+                );
+            else
+                env += extrapolate(
+                    TIME_2, 
+                    AMP_2, 
+                    TIME_3, 
+                    AMP_3, 
+                    runTime
+                    );
+        }
+        else if (not after_peakpulse) 
+            env = AMPLITUDE; 
+	else // after startDownramp
+            env = AMPLITUDE * gauss(runTime-startDownramp);
+        return env;
+    }
+
+    HINLINE float3_X laserLongitudinal(uint32_t currentStep, float_X& phase)
+    {
+        float_X envelope;
+        float3_X elong(float3_X::create(0.0));
+
+        // a symmetric pulse will be initialized at position z=0 for
+        // a time of RAMP_INIT * PULSE_LENGTH + LASER_NOFOCUS_CONSTANT = INIT_TIME.
+        // we shift the complete pulse for the half of this time to start with
+        // the front of the laser pulse.
+
+        /* initialize the laser not in the first cell is equal to a negative shift
+         * in time
+         */
+        const float_64 runTime = (DELTA_T * currentStep - laserTimeShift -
+            0.5 * RAMP_INIT * PULSE_LENGTH);
+
+        const float_64 tau = PULSE_LENGTH * sqrt(2.0);
+
+        phase += float_X(w * runTime) + LASER_PHASE ;
+
+        envelope = get_envelope(runTime);
+
+
+        if( Polarisation == LINEAR_X )
+        {
+            elong.x() = float_X(envelope * (math::sin(phase)));
+        }
+        else if( Polarisation == LINEAR_Z )
+        {
+            elong.z() = float_X(envelope * (math::sin(phase)));
+        }
+        else if( Polarisation == CIRCULAR )
+        {
+            elong.x() = float_X(envelope / sqrt(2.0) * (math::sin(phase)));
+            elong.z() = float_X(envelope / sqrt(2.0) * (math::cos(phase)));
+        }
+
+        return elong;
+    }
+
+    /**
+     *
+     * @param elong
+     * @param phase
+     * @param posX
+     * @param posZ
+     * @return
+     */
+    HDINLINE float3_X laserTransversal(float3_X elong, const float_X, const float_X posX, const float_X posZ)
+    {
+
+        const float_X exp_x = posX * posX / (W0_X * W0_X);
+        const float_X exp_z = posZ * posZ / (W0_Z * W0_Z);
+
+        return elong * math::exp( float_X( -1.0 ) * ( exp_x + exp_z ) );
+
+    }
+
+}
+}
+
```

Comment:
please only one empty newline at the end of a file

Reply:
I removed the rest (do I have to write something here to get rid of the red mark near your comment?)